cloud/amazon/ec2_ami.py

```diff
@@ -214,11 +338,34 @@ def create_image(module, ec2):
         image_id = ec2.create_image(**params)
     except boto.exception.BotoServerError, e:
         if e.error_code == 'InvalidAMIName.Duplicate':
-            images = ec2.get_all_images()
+            if owners:
+        	# Don't need to loop over all images if dealing with just own ones
+        	# Improves response speed if owner specified
+        	images = ec2.get_all_images(owners=owners)
+    	    else:
+    		images = ec2.get_all_images()
             for img in images:
                 if img.name == name:
-                    module.exit_json(msg="AMI name already present", image_id=img.id, state=img.state, changed=False)
-                    sys.exit(0)
+		    module.exit_json(
+			msg = "AMI name already present",
+			image_id = img.id,
+			state = img.state,
+    			changed = False,
+    			architecture = img.architecture,
+    			block_device_mapping = get_block_device_mapping(img),
+    			creationDate = img.creationDate,
+    			description = img.description,
+    			hypervisor = img.hypervisor,
+			is_public = img.is_public,
+			location = img.location,
+			name = img.name,
+			ownerId = img.ownerId,
+			platform = img.platform,
+			root_device_name = img.root_device_name,
+    			root_device_type = img.root_device_type,
+    			tags = img.tags,
+			virtualization_type = img.virtualization_type
+		    )
```

Comment:
I would make this a function call so you can reference it in your other exit_json statement below (DRY principle).

E.g.

def get_ami_info(img) {
 ...
}

Then have ...

exit_json(msg="bla", changed=True, get_ami_info(img))